debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/AbstractReplicationMessageColumn.java

```diff
@@ -105,30 +103,50 @@ public TypeMetadataImpl(String columnName, String typeWithModifiers, boolean opt
             }
             String normalizedTypeName = PgOid.normalizeTypeName(baseType);
 
+            if (schema != null) {
+                // strip . suffix
+                schema = schema.substring(0, schema.length()-1);
+            }
+
             if (isArray) {
                 normalizedTypeName = "_" + normalizedTypeName;
             }
             this.baseType = baseType;
             this.fullType = fullType;
             this.normalizedTypeName = normalizedTypeName;
+            this.schema = schema;
 
+            // TODO: make this more elegant/type-specific
             if (typeModifiers.length > 0) {
-                this.length = typeModifiers[0];
+                try {
+                    this.length = Integer.parseInt(typeModifiers[0]);
+                } catch (NumberFormatException e) {
```

Comment:
We should re-throw this (as before). Same below.

Reply:
We can't assume modifiers are numbers, and IMO it's a bad assumption that the first one is always a "length" and the second is a "scale".

eg. `CREATE TABLE t (mygeom geometry(MultiPolygon, 12345));` leads to `typeModifiers={"MultiPolygon", "12345"}` — the second modifier is a coordinate system identifier, not a scale.